server.js

```diff
@@ -0,0 +1,278 @@
+    path: '/hydrogen.html',
+    contentType: `text/html`,
+    content: `<!DOCTYPE html>
+    <html lang="en">
+    <head>
+      <meta charset="UTF-8">
+      <title>The Elements - Hydrogen</title>
+      <link rel="stylesheet" href="/css/styles.css">
+    </head>
+    <body>
+      <h1>Hydrogen</h1>
+      <h2>H</h2>
+      <h3>Atomic number 1</h3>
+      <p>Hydrogen is a chemical element with chemical symbol H and atomic number 1. With an atomic weight of 1.00794 u, hydrogen is the lightest element on the periodic table. Its monatomic form (H) is the most abundant chemical substance in the universe, constituting roughly 75% of all baryonic mass. Non-remnant stars are mainly composed of hydrogen in its plasma state. The most common isotope of hydrogen, termed protium (name rarely used, symbol 1H), has a single proton and zero neutrons.</p>
+      <p><a href="/">back</a></p>
+    </body>
+    </html>`
+  },
+  {
+    method: 'GET',
+    statusCode: '200 OK',
+    path: '/helium.html',
+    contentType: `text/html`,
+    content: `<!DOCTYPE html>
+    <html lang="en">
+    <head>
+      <meta charset="UTF-8">
+      <title>The Elements - Helium</title>
+      <link rel="stylesheet" href="/css/styles.css">
+    </head>
+    <body>
+      <h1>Helium</h1>
+      <h2>H</h2>
+      <h3>Atomic number 2</h3>
+      <p>Helium is a chemical element with symbol He and atomic number 2. It is a colorless, odorless, tasteless, non-toxic, inert, monatomic gas that heads the noble gas group in the periodic table. Its boiling and melting points are the lowest among all the elements and it exists only as a gas except in extremely cold conditions.</p>
+      <p><a href="/">back</a></p>
+    </body>
+    </html>`
+  },
+  {
+    method: 'GET',
+    statusCode: '404 Not Found',
+    path: '/404.html',
+    contentType: `text/html`,
+    content: `<!DOCTYPE html>
+    <html lang="en">
+    <head>
+      <meta charset="UTF-8">
+      <title>Element not found!</title>
+      <link rel="stylesheet" href="/css/styles.css">
+    </head>
+    <body>
+      <h1>404</h1>
+      <h2>Element not found!</h2>
+      <p>
+        <a href="/">back</a>
+      </p>
+    </body>
+    </html>`
+  }
+]
+
+// this creates a server
+const server = net.createServer((socket) => {
+  socket.setEncoding('utf8');
+  // same code is repeated per request, such as index and dependencies like css.
+  socket.on('data', (data) => {
+    console.log(data);
+    // getting the path or uri of the http request
+    let uri = data.slice(data.indexOf('/'));
+    uri = uri.slice(0, uri.indexOf(' '));
+    // finding the route to the page specified
+    let page = routes.filter((route) => {
+      if (route.path === uri){
+        return route;
+      }
+    }, undefined)[0];
+    // determining whether to serve page if found in routes, or 404 page if page variable
+    // is undefined.
+    let response;
+    if (page){
+      buildResponse(page);
+    } else {
+      buildResponse(routes[5]);
```

Comment:
routes[5] is not as descriptive as something like a variable called notFoundPage.